### PR TITLE
Add active state to Popover2 when popover is open and target is a button

### PR DESCRIPTION
**Pull request by gjavitt**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [ ] Include tests
- [ ] Update documentation

#### Changes proposed in this pull request:
Adds active state to Popover2 when popover is open and target is a button. (Does not make button active if popover is triggered by hover interaction instead of click.)

#### Screenshot
image

### CODE REVIEW

File: packages/labs/src/components/popover/popover2.tsx [hunk 1]
```diff
@@ -288,9 +289,10 @@ export class Popover2 extends AbstractComponent<IPopover2Props, IPopover2State>
         );
 
         const children = this.understandChildren();
-        const targetTabIndex = this.props.openOnTargetFocus && this.isHoverInteractionKind() ? 0 : undefined;
+        const targetTabIndex = this.props.openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
         const target: JSX.Element = React.cloneElement(children.target, {
             // force disable single Tooltip child when popover is open (BLUEPRINT-552)
+            className: classNames({ [Classes.ACTIVE]: isOpen && !isHoverInteractionKind }),
```
Comment (cmslewis) [inline packages/labs/src/components/popover/popover2.tsx]: Need to make sure we don't overwrite custom target classes that might have been provided. Let's move this to the `classNames` composition on [L282](https://github.com/palantir/blueprint/pull/1872/files#diff-affb716df70db56e0ecd5eecaae135abR283):

```tsx
targetProps.className = classNames(
    Classes.POPOVER_TARGET,
    {
        [Classes.ACTIVE]: isOpen && !isHoverInteractionKind,
        [Classes.POPOVER_OPEN]: isOpen,
    },
    className,
);
```

Comment (llorca) [reply packages/labs/src/components/popover/popover2.tsx]: @cmslewis `targetProps.className` or `target.className`?

Comment (cmslewis) [reply packages/labs/src/components/popover/popover2.tsx]: `targetProps`. Here's the code that's already present:

image

**Comment by adidahiya** [comment]
@gjavitt please merge master into your branch to pick up the new CI workflows

**Comment by llorca** [comment]
@giladgray is this PR cool? if not, can you think of a better approach?

File: packages/labs/src/components/popover/popover2.tsx [hunk 2]
```diff
@@ -288,8 +289,9 @@ export class Popover2 extends AbstractComponent<IPopover2Props, IPopover2State>
         );
 
         const children = this.understandChildren();
-        const targetTabIndex = this.props.openOnTargetFocus && this.isHoverInteractionKind() ? 0 : undefined;
+        const targetTabIndex = this.props.openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
         const target: JSX.Element = React.cloneElement(children.target, {
+            className: classNames({ [Classes.ACTIVE]: isOpen && !isHoverInteractionKind }),
```
Comment (giladgray) [inline packages/labs/src/components/popover/popover2.tsx]: must add `children.target.props.className` to avoid clobbering existing classes.

**Comment by giladgray** [comment]
@gjavitt can you please [enable preview comments](https://github.com/palantir/blueprint/blob/master/CONTRIBUTING.md#enable-preview-comments) for your Circle build so we can play with it?

**Comment by gjavitt** [comment]
<h3>unclobber classnames</h3>

Preview: <strong><a href='https://11-113383053-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html' target='_blank'>documentation</a> | <a href='https://11-113383053-gh.circle-artifacts.com/0/home/circleci/project/packages/landing-app/dist/index.html' target='_blank'>landing</a> | <a href='https://11-113383053-gh.circle-artifacts.com/0/home/circleci/project/packages/table-dev-app/dist/index.html' target='_blank'>table</a></strong>

**Comment by llorca** [comment]
Agree Gilad, if anything it's something we should fix on the `Tooltip`/`popover-target` side, it's biting us in many places

**Pull request merged**